### ISSUE TITLE
exiftool should be configurable externally

### ISSUE THREAD

**Issue by pbranly** [nemethviktor/GeoTagNinja#11]
it seems that the version of exiftool is 3 months old.
as in the geosetter program the exiftool executable should be modified externally as it evolves very often.
Current situation is that it is compiled with the whole package

For a future release.... not urgent

Phil

**Comment by nemethviktor** [comment]
Yeah, aware, vaguely. There's a part of the code that checks for new exifTool version but for some reason Phil hasn't uploaded the current version to https://metacpan.org/pod/Image::ExifTool and I haven't managed to find another place that has an API that can be queried...hence GTN still thinks the 3-months-old version is "current". 

I'll add in an option for an external path, that shouldn't take long :)

**Comment by nemethviktor** [comment]
Indeed. But his site doesn't have a queryable API for the newest version hence I'm using metacpan.... I'll drop a msg in the exiftool forum about this and see if they can recommend a better site.

**Comment by nemethviktor** [comment]
Actually had a 2nd thought on this. I'm tempted not to make this enhancement: for the app to work properly there needs to be an .ExifTool_config file in the folder where exiftool (ET) itself is running. That file needs to have a very precise content in it otherwise the app will crash (and it's a pretty important config thing that i can't otherwise enforce). This poses a number of issues:
1. the user that has an external path to ET wouldn't have the config file unless they copy it and 
2. there may be other apps they use that require different configs.
3. the GTN app might not have write permissions to just copy into a "random" folder (somewhat to offset issue 1), ie Program Files is typically one of those places that I've had problems with and yet that'd be the most likely location for an external ET folder (from what I remember apps can't really write into Program Files but can read from it, hence the database.sqlite file sitting entirely elsewhere).
4. users aren't generally techy enough to go and hunt for random existing files in random existing folders (thinking of asking Everyday Joe to copy the .ExifTool_config from GTN's install/resources folder to whereever else).

The TLDR is that this is a bit too complicated to make fool-proof.

StarGeek has kindly [suggested](https://exiftool.org/forum/index.php?topic=14086.0)  a more up-to-date way of trying to find the newest exiftool and I can change the code to only alert when the newest version is newer than the current one. (bcs atm it warns if it's different, even if older, ie user could get 12.49 from exiftool.org but the app would meow that on the API that's in the current code it's 12.42) - this way if there was a new ET version users that know how to do it can copy the file into the relevant path. At the same time instead of nagging users each time they launch GTN there's a new ET version it would only be done once.

Thoughts?

**Comment by nemethviktor** [comment]
Fixed in today's release. :)

**Comment by pbranly** [comment]
Hi, this is not clear for me:
I have upgraded GTN to the last release
then I started the application and got a new message that I did not read
(!) as I thought it asked again for a new release.
Then I found the last version of exiftool in zip format in my download
directory
I am lost now :
Do I have to copy and paste the exiftool.exe (without the -k for sure) in
the C:\Program Files\GeoTagNinja\Resources directory ?
it seems that the one that is inside that directory is dated 10th Aug 2022
and so is not the last one
And when I start again GTN I dont have any new message
Phil

Le mar. 25 oct. 2022 à 12:08, nemethviktor ***@***.***> a
écrit :

**Comment by nemethviktor** [comment]
Yes I admittedly forgot to include the new exiftool version in the release - the code change was for the following:
- check through https://exiftool.org/ver.txt what the newest version is
- check locally if the exiftool file (that's likely in C:\Program Files\GeoTagNinja\Resources) is the same version
- if yes, don't do anything
- if no, warn only once (not each time) that there is a new version. if the user wants then they can download, unzip and copy to C:\Program Files\GeoTagNinja\Resources (but rename "(-k)")

**Comment by pbranly** [comment]
clear ! I understood the procedure while making the translation
